services/api/src/utils/middleware/authenticate.js

```diff
@@ -19,22 +24,22 @@ function getToken(ctx) {
 function validateToken(ctx, token, type) {
   // ignoring signature for the moment
   const decoded = jwt.decode(token, { complete: true });
-  if (decoded === null) return ctx.throw(401, 'bad jwt token');
+  if (decoded === null) throw new TokenError('bad jwt token');
   const { payload } = decoded;
```

Comment:
don't get hung up on that it, the main point is that these are defined in one place and doesn't require a lot of knowledge about the details etc. It could be a const as well ie `ctx.throw(401, 'bad jwt token', TOKEN_ERROR)` or some thing but I find that much more obtuse.